apps/typegpu-docs/src/content/examples/simple/function-visualizer/index.ts

```diff
@@ -0,0 +1,532 @@
+    });
+    pass.setPipeline(renderPipeline);
+    pass.setBindGroup(0, root.unwrap(renderBindGroup));
+    // call our vertex shader 2 times per point drawn
+    pass.draw(properties.interpolationPoints * 2);
+  });
+  pass.end();
+
+  device.queue.submit([encoder.finish()]);
+}
+
+// Helper definitions
+
+function fromHex(hex: string) {
+  const r = Number.parseInt(hex.slice(1, 3), 16);
+  const g = Number.parseInt(hex.slice(3, 5), 16);
+  const b = Number.parseInt(hex.slice(5, 7), 16);
+
+  return d.vec4f(r / 255.0, g / 255.0, b / 255.0, 1.0);
+}
+
+async function tryRecreateComputePipeline(
+  functionCode: string,
+): Promise<GPUComputePipeline> {
+  const codeToCompile = functionCode === '' ? '0' : functionCode;
+
+  const computeShaderCode = createComputeShaderCode(codeToCompile);
+
+  device.pushErrorScope('validation');
+  const computeShaderModule = device.createShaderModule({
+    label: `Compute function points shader module for f(x) = ${codeToCompile}`,
+    code: computeShaderCode,
+  });
+  const error = await device.popErrorScope();
+  if (error) {
+    throw new Error(`Function f(x) = '${codeToCompile}' is invalid.`);
+  }
+
+  const computePipeline = device.createComputePipeline({
+    label: 'Compute function points pipeline',
+    layout: device.createPipelineLayout({
+      bindGroupLayouts: [root.unwrap(computeLayout)],
+    }),
+    compute: {
+      module: computeShaderModule,
+    },
+  });
+
+  return computePipeline;
+}
+
+function queuePropertiesBufferUpdate() {
+  properties.inverseTransformation = mat4.inverse(properties.transformation);
+  propertiesBuffer.write(properties);
+}
+
+// Canvas controls
+
+let lastPos: number[] | null = null;
+
+canvas.onmousedown = (event) => {
+  lastPos = [event.offsetX, event.offsetY];
+};
+
+canvas.onmouseup = (_) => {
+  lastPos = null;
+};
+
+canvas.onmousemove = (event) => {
+  if (lastPos == null) {
+    return;
+  }
+  const currentPos = [event.offsetX, event.offsetY];
+  const translation = [
+    (-(currentPos[0] - lastPos[0]) / canvas.width) *
+      2.0 *
+      window.devicePixelRatio,
+    ((currentPos[1] - lastPos[1]) / canvas.height) *
+      2.0 *
+      window.devicePixelRatio,
+    0.0,
+  ];
+  mat4.translate(
+    properties.transformation,
+    translation,
+    properties.transformation,
+  );
+
+  lastPos = currentPos;
+};
+
+canvas.onwheel = (event) => {
+  event.preventDefault();
+
+  const delta = Math.abs(event.deltaY) / 1000.0 + 1;
+  const scale = event.deltaY > 0 ? delta : 1 / delta;
+
+  mat4.scale(
+    properties.transformation,
+    [scale, scale, 1],
+    properties.transformation,
+  );
+};
+
+// #region Example controls and cleanup
+
+export const controls = {
+  [initialFunctions[0].name]: {
+    initial: initialFunctions[0].code,
+    onTextChange: async (value: string) => {
+      try {
+        computePipelines[0] = await tryRecreateComputePipeline(value);
+      } catch (e) {
+        console.log(e);
+      }
+    },
```

Comment:
For functions longer than one-liners, lets use this syntax.
```suggestion
    async onTextChange(value: string) {
      try {
        computePipelines[0] = await tryRecreateComputePipeline(value);
      } catch (e) {
        console.log(e);
      }
    },
```